core/src/main/java/com/opensymphony/xwork2/ognl/OgnlUtil.java

```diff
@@ -78,14 +78,45 @@
     private boolean allowStaticMethodAccess;
     private boolean disallowProxyMemberAccess;
 
+    /**
+     * Construct a new OgnlUtil instance for use with the framework
+     */
     public OgnlUtil() {
+        this(null, null);  // Instantiate default Expression and BeanInfo caches (null factories)
+    }
+
+    /**
+     * Construct a new OgnlUtil instance for use with the framework, with optional
+     * cache factories for Ognl Expression and BeanInfo caches.
+     * 
+     * @param ognlExpressionCacheFactory factory for Expression cache instance.  If null, use default
+     * @param ognlBeanInfoCacheFactory  for BeanInfo cache instance.  If null, use default
+     */
+    @Inject
+    public OgnlUtil(
+            @Inject(value = StrutsConstants.STRUTS_OGNL_EXPRESSIONCACHE_FACTORY, required = false) OgnlCacheFactory<String, Object> ognlExpressionCacheFactory,
+            @Inject(value = StrutsConstants.STRUTS_OGNL_BEANINFOCACHE_FACTORY, required = false) OgnlCacheFactory<Class<?>, BeanInfo> ognlBeanInfoCacheFactory
+    ) {
         excludedClasses = Collections.unmodifiableSet(new HashSet<>());
         excludedPackageNamePatterns = Collections.unmodifiableSet(new HashSet<>());
         excludedPackageNames = Collections.unmodifiableSet(new HashSet<>());
 
         devModeExcludedClasses = Collections.unmodifiableSet(new HashSet<>());
         devModeExcludedPackageNamePatterns = Collections.unmodifiableSet(new HashSet<>());
         devModeExcludedPackageNames = Collections.unmodifiableSet(new HashSet<>());
+        this.ognlExpressionCacheFactory = ognlExpressionCacheFactory;
+        this.ognlBeanInfoCacheFactory = ognlBeanInfoCacheFactory;
+
+        if (ognlExpressionCacheFactory != null) {
+            this.expressionCache = ognlExpressionCacheFactory.buildOgnlCache(ognlExpressionCacheFactory.getCacheMaxSize(), 16, 0.75f, ognlExpressionCacheFactory.getUseLRUCache());
+        } else {
+            this.expressionCache = new OgnlDefaultCache<>(25000, 16, 0.75f);
+        }
+        if (ognlBeanInfoCacheFactory != null) {
+            this.beanInfoCache = ognlBeanInfoCacheFactory.buildOgnlCache(ognlBeanInfoCacheFactory.getCacheMaxSize(), 16, 0.75f, ognlBeanInfoCacheFactory.getUseLRUCache());
+        } else {
+            this.beanInfoCache = new OgnlDefaultCache<>(25000, 16, 0.75f);
```

Comment:
Same here, `OgnlExpressionCacheFactory#createDefaultCache()`?

Reply:
Please see the comment for the expression cache item above.